core/relayMsgs.go

```diff
@@ -70,24 +76,39 @@ func (r *RelayMsgs) Send(src, dst Chain) {
 
 		if r.IsMaxTx(msgLen, txSize) {
 			// Submit the transactions to src chain and update its status
-			r.Succeeded = r.Succeeded && SendCheckMsgs(src, msgs)
-
+			msgIDs, err := src.SendMsgs(msgs)
+			if err != nil {
+				logger.Error("failed to send msgs", err, "msgs", msgs)
+			}
+			r.Succeeded = r.Succeeded && (err == nil)
+			for i := range msgs {
+				srcMsgIDs[i+srcMaxTxCount] = msgIDs[i]
+			}
 			// clear the current batch and reset variables
+			srcMaxTxCount += len(msgs)
 			msgLen, txSize = 1, uint64(len(bz))
 			msgs = []sdk.Msg{}
 		}
 		msgs = append(msgs, msg)
 	}
 
 	// submit leftover msgs
-	if len(msgs) > 0 && !SendCheckMsgs(src, msgs) {
-		r.Succeeded = false
+	if len(msgs) > 0 {
+		msgIDs, err := src.SendMsgs(msgs)
+		if err != nil {
+			logger.Error("failed to send msgs", err, "msgs", msgs)
+		}
+		r.Succeeded = r.Succeeded && (err == nil)
+		for i := range msgs {
+			srcMsgIDs[i+srcMaxTxCount] = msgIDs[i]
+		}
 	}
 
 	// reset variables
 	msgLen, txSize = 0, 0
 	msgs = []sdk.Msg{}
 
+	dstMaxTxCount := 0
```

Comment:
The variable used for src can be reused.

Reply:
https://github.com/hyperledger-labs/yui-relayer/commit/286bfe5359804373830a090058994c3dbbb0447b
Fixed